### DISCUSSION TITLE
new nilsimsa options

### DISCUSSION THREAD

**Discussion by hackernoodle** [mhx/dwarfs#190] [General]
I've been using the same set of options for a long time and no longer remember what they meant, but now in 0.8.0 there has been a change that makes the arguments invalid, so what is the new equivalent of `--order=nilsamsa:255:40000:40000`? clearly nilsimsa with 3 options, but the current manpage only mentions two.

**Comment by hackernoodle** [comment]
so i looked at the older manpage and it comes back to me, 255=limit and 40000=depth, how does that translate to max-children and max-cluster-size

**Reply by mhx** [reply]
It doesn't translate, sorry.

It's a completely new algorithm that does away with some of the shortcomings of the old version. The new version produces strictly repeatable results with the same options and can run multi-threaded. The old version was single-threaded and for the most part non-deterministic.

If you want "better" ordering than the default, try increasing the numbers for both options until the sorting step gets too slow for you.

**Reply by hackernoodle** [reply]
Ok, I can work with that. btw what are the default values max-childern and max-cluster-size?

**Reply by mhx** [reply]
It's 16384 for both values. I need to add that to the help/docs, sorry.

**Reply by hackernoodle** [reply]
am i doing something wrong? `--order=nilsimsa` and `--order=nilsimsa:max-children=100000:max-cluster-size=100000` take the same amount of time and produce images of the same size

**Reply by mhx** [reply]
How many files are in your input? Once the values are larger than the number of files, there's going to be no difference. So this is really only relevant if you try to pack lots of files.

**Reply by hackernoodle** [reply]
Oh! right, of course. nevermind then
